src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Receivers.kt

```diff
@@ -181,36 +261,35 @@ class MarkdownTaskReceiver(
 
     private fun getCmdLineArguments(request: CompileRequest): Map<String, String> {
         val outputFileName = Files.getNameWithoutExtension(request.mainFileName) + ".tex"
-        val projTasks = this.tasksDir.resolve(request.id)
-        val mainFile = request.mainFileName
-        val outputFile = projTasks.resolve(outputFileName)
-        val projectRootAbsPath = this.tasksDir.toAbsolutePath().parent
+        val projTasks = this.tasksDir.resolve(request.id).resolve("files")
+        val mainFile = StringUtils.stripStart(request.mainFileName, "/")
+        val values = listOf(projTasks.toString(), "",
+                mainFile, outputFileName, PANDOC_DEFAULT_FONT)
 
-        val values = listOf(projectRootAbsPath.toString(), "",
-                mainFile.toString(), outputFile.toString(), PANDOC_DEFAULT_FONT)
-
-        return (arguments zip values).map { it.first to it.second }.toMap()
+        return (arguments zip values).map { it.first to "\"${it.second}\"" }.toMap()
     }
 
     // converts Markdown into tex via pandoc
     private fun convertMarkdown(commandArguments: Map<String, String>): Int {
         val substitutor = StringSubstitutor(commandArguments)
-        val rawCommandLine = config.getString(COMPILE_COMMAND_KEY)
+        val rawCommandLine = config.getString(COMPILE_COMMAND_KEY).replace("\\", "")
```

Comment:
Can you add a comment about this "unescaping"? Also, the issue is with `${placeholderName}` constructs, right? Maybe restrict ourselves to `\\$` ?

Reply:
This is happening due to the double slash in the config. One `\` is existing after substituting.

Reply:
>  Is my understanding correct that the only such use case is escaping $

Yes, it comes from here.
I'm not sure that it is only case of this, because we have only such case. 

Do I understand correct that, your question is about general situation, that do we need always remove this backslash or only in particular with `$`?

Because we have only one line in the config I just left here as it is.